Thanks, tried to compile this but it's complaining:

`C:\...\Arduino\libraries\SI7021\SI7021.cpp: In member function 'void SI7021::setPrecision(byte)':
C:\...\Arduino\libraries\SI7021\SI7021.cpp:161: error: invalid conversion from 'int' to 'byte*'
C:\...\Arduino\libraries\SI7021\SI7021.cpp:161: error:   initializing argument 1 of 'void SI7021::_writeReg(byte*, int)'`

--- a/SI7021.cpp
+++ b/SI7021.cpp
@@ -140,6 +140,20 @@ int SI7021::getDeviceId() {
   return id;
 }
 
+// 00 = 14 bit temp, 12 bit RH
+// 01 = 12 bit temp, 8 bit RH
+// 10 = 13 bit temp, 10 bit RH (default)
+// 11 = 11 bit temp, 11 bit RH
+void SI7021::setPrecision(byte setting) {
+    byte reg = USER1_READ;
+    _writeReg(&reg, 1);
+    _readReg(&reg, 1);
+
+    reg = (reg & 0xFC) | (setting & 3);
+    byte userwrite[] = {USER1_WRITE, reg};
+    _writeReg(userwrite, sizeof userwrite);
+}
+
 void SI7021::setHeater(bool on) {
     byte userbyte;
     if (on) {

--- a/SI7021.h
+++ b/SI7021.h
@@ -54,6 +54,7 @@ class SI7021
     struct si7021_thc getTempAndRH();
     int getSerialBytes(byte * buf);
     int getDeviceId();
+    void setPrecision(byte setting);
     void setHeater(bool on);
   private:
     void _command(byte cmd, byte * buf );

